src/app/timer-card/timer-card.component.ts

```diff
@@ -1,22 +1,82 @@
-import { Component } from '@angular/core';
+import { Component, Input, AfterViewInit } from '@angular/core';
+import {
+  Observable,
+  timer,
+  tap,
+  takeWhile,
+  fromEvent,
+  map,
+  withLatestFrom,
+  scan,
+  filter,
+} from 'rxjs';
 
+export type Status = {
+  active: boolean;
+};
 @Component({
   selector: 'app-timer-card',
   templateUrl: './timer-card.component.html',
   styleUrls: ['./timer-card.component.scss'],
 })
-export class TimerCardComponent {
-  timeInSeconds: number = 585;
-  label: string = 'Label';
-  isPaused: Boolean = false;
-  color: string = 'warn';
+export class TimerCardComponent implements AfterViewInit {
+  @Input() timeInSeconds!: number;
+  @Input() id!: number;
+  @Input() label!: string;
+  color: string = 'primary';
+  isPaused: Boolean = true;
+  isCompleted: Boolean = false;
+  progress: number = 100;
+  countdown$!: Observable<number>;
+  private pauseClick$!: Observable<Status>;
+  private tick$ = timer(0, 1000);
+
+  ngAfterViewInit(): void {
+    this.pauseClick$ = fromEvent(
+      document.getElementById('' + this.id) as HTMLButtonElement,
+      'click'
+    ).pipe(
+      map(() => ({
+        active: !this.isPaused,
+      }))
+    );
+    this.countdown$ = this.timer(this.timeInSeconds);
+  }
 
   pauseTimer() {
     this.isPaused = !this.isPaused;
     this.color = this.isPaused ? 'primary' : 'warn';
-    //functionality will be added with the rxjs streams implementation.
   }
+
   deleteTimer() {
     //functionality will be added with the rxjs streams implementation.
   }
+
+  reloadTimer() {
+    this.isCompleted = false;
+    this.progress = 100;
+    if (!this.isPaused) this.pauseTimer();
+    this.countdown$ = this.timer(this.timeInSeconds);
+  }
+
+  timer(value: number): Observable<number> {
+    return this.tick$.pipe(
```

Comment:
I would separate both streams - 1 would be for counting down, and the other for - derived from the previous for transforming it into 0-100 scale.